City-of-Helsinki/hel-fi-drupal-grants
# feat: AU-2014: E2E:  Draft submission deletion tests

# [AU-2014](https://helsinkisolutionoffice.atlassian.net/browse/AU-2014)

## What was done
This pull request implements draft application deletion tests. All application variants that are saved as `DRAFT` (`wrong_values`, `missing_values`, `draft`) are deleted using this approach. Applications are randomly deleted in one of two ways:

1. By going to an applications `submission URL` (e.g. `/fi/hakemus/liikunta_toiminta_ja_tilankaytto/1391/muokkaa`) and clicking the delete button there. This is handled by the `deleteUsingSubmissionUrl` function.

2. By going to the `application listing` on the `Oma asiointi page` and clicking the delete button there. Here the application ID is used to locate the application. This is handled by the `deleteUsingApplicationId` function.

## How to install
Make sure your instance is up and running on the correct branch.
  * `git checkout feature/AU-2014-e2e-application-deletion`
  * `make fresh`
  * `make drush-cr`

## How to test

- Review `deletion_helpers.ts`

- Run a few tests, like `make test-pw registered_community_53`, `make test-pw registered_community_63`, and `make test-pw-p PROJECT=forms-48`.

- Make sure you get an ouput that looks similar to this:

![Screenshot 2024-02-07 at 14 31 06](https://github.com/City-of-Helsinki/hel-fi-drupal-grants/assets/117807015/9a93ef54-98ba-4c79-9200-80d9c4084e52)


[AU-2014]: https://helsinkisolutionoffice.atlassian.net/browse/AU-2014?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
